src/Core/functions.php

```diff
@@ -526,3 +531,88 @@ function toBool(mixed $value): ?bool
 
     return null;
 }
+
+/**
+ * Converts a value to a native DateTime object.
+ *
+ *  integer - value is treated as a Unix timestamp
+ *  string - value is treated as a ISO-8601 (Atom) formatted timestamp
+ *  Other values returns as null.
```

Comment:
I like this direction. I wonder if we could also use the default formatting types that `Database\Type\DateTimeType` uses. A potentially simpler solution would be to allow users to provide a format to parse with and default the parameter to the `ATOM` format, or ISO-8601 format.